tensorboard/pip_package/pip_smoke_test.sh

```diff
@@ -0,0 +1,172 @@
+echo "Settings:"
+echo "  PY_VERSION=${PY_VERSION}"
+echo "  TEST_PORT=${TEST_PORT}"
+echo "  NUM_RETRIES=${NUM_RETRIES}"
+echo
+
+# Check that virtualenv is installed.
+if [[ -z "$(which virtualenv)" ]]; then
+  die "ERROR: virtualenv is required, but does not appear to be installed."
+fi
+
+PIP_TMP_DIR=$(mktemp -d --suffix _tensorboard)
+
+echo
+echo "Building tensorboard pip package in directory: ${PIP_TMP_DIR}"
+echo
+
+bazel build tensorboard/pip_package:build_pip_package
+
+# Create virtualenv directory, cleanly (i.e., no --system-site-packages).
+VENV_TMP_DIR=$(mktemp -d --suffix _tensorboard_venv)
+
+echo
+echo "Creating virtualenv directory at: ${VENV_TMP_DIR}"
+echo
+
+if [[ ${PY_VERSION} == 2 ]]; then
+  virtualenv -p python "${VENV_TMP_DIR}"
+  TF_NIGHTLY_URL='https://ci.tensorflow.org/view/Nightly/job/nightly-matrix-cpu/TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=PIP,TF_BUILD_PYTHON_VERSION=PYTHON2,label=cpu-slave/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-1.head-cp27-none-linux_x86_64.whl'
+elif [[ ${PY_VERSION} == 3 ]]; then
+  virtualenv -p python3 "${VENV_TMP_DIR}"
+  TF_NIGHTLY_URL='https://ci.tensorflow.org/view/Nightly/job/nightly-matrix-cpu/TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=PIP,TF_BUILD_PYTHON_VERSION=PYTHON3,label=cpu-slave/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-1.head-cp34-cp34m-linux_x86_64.whl'
+fi
+
+echo
+echo "Activating virtualenv at ${VENV_TMP_DIR}"
+echo
+
+export VIRTUAL_ENV="${VENV_TMP_DIR}"
+export PATH="${VENV_TMP_DIR}/bin:${PATH}"
+unset PYTHON_HOME
+
+echo
+echo "Performing pip required for building wheels"
+echo
+
+pip install upgrade pip setuptools wheel
+
+echo
+echo "Creating tensorboard pip package in directory: ${PIP_TMP_DIR}"
+echo
+
+bazel-bin/tensorboard/pip_package/build_pip_package "${PIP_TMP_DIR}"
+
+# Install the dependency, tensorflow, first.
+echo
+echo "Installing nightly tensorflow pip package."
+echo
+
+pip install ${TF_NIGHTLY_URL}
+
+echo
+echo "Installing the just-built tensorboard pip package"
+echo
+
+if [[ ${PY_VERSION} == 2 ]]; then
+  pip install "${PIP_TMP_DIR}"/tensorflow_tensorboard*-py2-*.whl
+elif [[ ${PY_VERSION} == 3 ]]; then
+  pip install "${PIP_TMP_DIR}"/tensorflow_tensorboard*-py3-*.whl
+fi
+
+# Check tensorboard binary path.
+TB_BIN_PATH=$(which tensorboard)
+if [[ -z ${TB_BIN_PATH} ]]; then
+  die "ERROR: Cannot find tensorboard binary path after installing tensorboard pip package."
+fi
+
+TMP_LOGDIR=$(mktemp -d --suffix _tensorboard_logdir)
+tensorboard --port="${TEST_PORT}" --logdir="${TMP_LOGDIR}" &
+TB_PID=$!
+
+TEST_URL="http://localhost:${TEST_PORT}/data/logdir"
+echo
+echo "tensorboard binary should be running at pid ${TB_PID}"
+echo "Sending test HTTP requests at URL: ${TEST_URL} (${NUM_RETRIES} retries)"
+echo
+
+RETRY_COUNTER=0
+REQUEST_SUCCEEDED=0
+while [[ ${RETRY_COUNTER} -lt ${NUM_RETRIES} ]]; do
+  sleep 1
+  set +e
+  STATUS_CODE=$(curl -Is "${TEST_URL}" | head -1 | cut -d ' ' -f 2)
```

Comment:
Done. There was another issue in the code that `set -e` masked. Now that issue is gone, we can get rid of the `set -e`. Done.